src/main/java/core/basesyntax/OperationsStrategyImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax;
+
+public class OperationsStrategyImpl implements OperationStrategy {
+    private final Fruits fruits;
+
+    public OperationsStrategyImpl(Fruits fruits) {
+        this.fruits = fruits;
+    }
+
+    public int operation(String operation, String fruitType, int amount) {
+        switch (operation) {
+            case "b": // Balance
+                return fruits.balance(fruitType, amount);
+            case "s": // Supply
+                return fruits.supply(fruitType, amount);
+            case "p": // Purchase
+                return fruits.purchase(fruitType, amount);
+            case "r": // Return
+                return fruits.returnFruit(fruitType, amount);
+            default:
+                throw new IllegalArgumentException("Unknown operation: " + operation);

```

Comment:
The exception message could be more informative by specifying the valid operations. Consider updating the message to something like: `"Unknown operation: " + operation + ". Valid operations are: b (balance), s (supply), p (purchase), r (return)."`